packages/esm-patient-conditions-app/src/conditions/conditions-form.component.tsx

```diff
@@ -1,479 +1,313 @@
-                            }}
-                          />
-                          <span>
-                            <Trans i18nKey="historyOf">History of</Trans>
-                          </span>
-                        </label>
-                      </div>
-                    </div>
-                  </div>
-                  {inactiveStatus && (
-                    <div className={styles.conditionsInputContainer}>
-                      <label htmlFor="dateOfInactivity">
-                        <Trans i18nKey="dateOfInactivity">
-                          Date of inactivity
-                        </Trans>
-                      </label>
-                      <div className="omrs-datepicker">
-                        <input
-                          type="date"
-                          id="dateOfInactivity"
-                          name="dateOfInactivity"
-                          defaultValue={inactivityDate}
-                          onChange={(event) =>
-                            setInactivityDate(event.target.value)
-                          }
-                        />
-                        <svg className="omrs-icon" role="img">
-                          <use xlinkHref="#omrs-icon-calendar"></use>
-                        </svg>
-                      </div>
-                    </div>
-                  )}
-                </div>
-              </div>
-            </SummaryCard>
-            <div
-              className={
-                enableEditFormButtons
-                  ? styles.buttonStyles
-                  : `${styles.buttonStyles} ${styles.buttonStylesBorder}`
-              }
-            >
-              <button
-                type="submit"
-                className="omrs-btn omrs-outlined-neutral omrs-rounded"
-                style={{ width: "20%" }}
-              >
-                <Trans i18nKey="delete">Delete</Trans>
-              </button>
-              <button
-                type="button"
-                className="omrs-btn omrs-outlined-neutral omrs-rounded"
-                onClick={closeForm}
-                style={{ width: "30%" }}
-              >
-                <Trans i18nKey="cancelChanges">Cancel changes</Trans>
-              </button>
-              <button
-                type="submit"
-                className={
-                  enableEditFormButtons
-                    ? "omrs-btn omrs-filled-action omrs-rounded"
-                    : "omrs-btn omrs-outlined omrs-rounded"
-                }
-                onClick={(event) => handleEditSubmit(event)}
-                disabled={!enableEditFormButtons}
-                style={{ width: "50%" }}
-              >
-                <Trans i18nKey="signAndSave">Sign & Save</Trans>
-              </button>
-            </div>
-          </form>
-        )}
-      </>
-    );
-  }
+  React.useEffect(() => {
+    debouncedSearch((viewState as SearchState)?.searchTerm);
+  }, [viewState, debouncedSearch]);
```

Comment:
The effect should depend only on the searchTerm and not on the viewState.
This does not seem like a big deal but you can get stuck in an infinite loop, where searchResults are fetched -> viewState is updated -> debounced effect is triggered -> searchResults are fetched -> ...

Reply:
Fixed. Sorry, must've missed it doing a late-night refactor!